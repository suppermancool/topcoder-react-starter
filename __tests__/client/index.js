import _ from 'lodash';

jest.mock('shared', () => ({
  default: 'APPLICATION',
}));

jest.mock('reducers', () => ({
  factory: () => () => ({}),
}));

jest.mock('topcoder-react-utils', () => {
  const TRU = require.requireActual('topcoder-react-utils');
  return {
    ...TRU,
    client: jest.fn(),
  };
});

test('Passes basic testing', () => {
  require('client');
  const { client } = require('topcoder-react-utils');

  const ops = client.mock.calls[0][0];
  expect(_.omit(ops, 'applicationModulePath')).toMatchSnapshot();

  expect(ops.applicationModulePath.endsWith('/src/shared/index.jsx'))
    .toBe(true);

  expect(ops.getApplication()).toMatchSnapshot();

  expect(ops.storeFactory()).toBeInstanceOf(Promise);
});
