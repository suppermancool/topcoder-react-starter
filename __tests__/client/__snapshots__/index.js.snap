// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Passes basic testing 1`] = `
Object {
  "getApplication": [Function],
  "moduleHot": undefined,
  "storeFactory": [Function],
}
`;

exports[`Passes basic testing 2`] = `[Function]`;
